misc/python/materialize/parallel_workload/parallel_workload.py

```diff
@@ -0,0 +1,302 @@
+
+    print(
+        f"--- Running with: --seed={seed} --threads={num_threads} --runtime={runtime} --complexity={complexity} --scenario={scenario} (--host={host} --port={port})"
+    )
+    initialize_logging()
+
+    system_conn = pg8000.connect(
+        host=host, port=system_port, user="mz_system", database="materialize"
+    )
+    system_conn.autocommit = True
+    with system_conn.cursor() as cur:
+        # TODO: Shuffle settings which should always work
+        cur.execute("ALTER SYSTEM SET enable_managed_clusters = true")
+        cur.execute("ALTER SYSTEM SET enable_rbac_checks = true")
+        cur.execute("ALTER SYSTEM SET enable_ld_rbac_checks = true")
+        cur.execute("ALTER SYSTEM SET persist_sink_minimum_batch_updates = 128")
+        cur.execute("ALTER SYSTEM SET enable_multi_worker_storage_persist_sink = true")
+        cur.execute("ALTER SYSTEM SET storage_persist_sink_minimum_batch_updates = 100")
+        cur.execute("ALTER SYSTEM SET persist_pubsub_push_diff_enabled = true")
+        cur.execute("ALTER SYSTEM SET persist_pubsub_client_enabled = true")
+        cur.execute("ALTER SYSTEM SET persist_stats_filter_enabled = true")
+        cur.execute("ALTER SYSTEM SET persist_stats_collection_enabled = true")
+        cur.execute("ALTER SYSTEM SET persist_stats_audit_percent = 100")
+        cur.execute("ALTER SYSTEM SET enable_monotonic_oneshot_selects = true")
+        cur.execute(
+            "ALTER SYSTEM SET persist_next_listen_batch_retryer_clamp = '100ms'"
+        )
+        cur.execute(
+            "ALTER SYSTEM SET persist_next_listen_batch_retryer_initial_backoff = '1200ms'"
+        )
+        cur.execute("ALTER SYSTEM SET max_schemas_per_database = 105")
+        cur.execute("ALTER SYSTEM SET max_tables = 105")
+        cur.execute("ALTER SYSTEM SET max_materialized_views = 105")
+        cur.execute("ALTER SYSTEM SET max_sources = 105")
+        cur.execute("ALTER SYSTEM SET max_roles = 105")
+        cur.execute("ALTER SYSTEM SET max_clusters = 105")
+        cur.execute("ALTER SYSTEM SET max_replicas_per_cluster = 105")
+
+    end_time = (
+        datetime.datetime.now() + datetime.timedelta(seconds=runtime)
+    ).timestamp()
+
+    rng = random.Random(random.randrange(SEED_RANGE))
+    database = Database(rng, seed, host, port, system_port, complexity, scenario)
+    conn = pg8000.connect(host=host, port=port, user="materialize")
+    conn.autocommit = True
+    with conn.cursor() as cur:
+        database.create(Executor(rng, cur))
+    conn.close()
+
+    conn = pg8000.connect(
+        host=host, port=port, user="materialize", database=str(database)
+    )
+    conn.autocommit = True
+    with conn.cursor() as cur:
+        database.create_relations(Executor(rng, cur))
+    conn.close()
+
+    workers = []
+    threads = []
+    for i in range(num_threads):
+        worker_rng = random.Random(rng.randrange(SEED_RANGE))
+        weights: List[float]
+        if complexity == "ddl":
```

Comment:
Ideally you would want the complexities to be an ENUM or a set of subclasses , and not strings, so that they are type-checked and misspellings are hopefully caught. E.g. what if this here was misspelled as `dll`, ?

Reply:
Yeah, absolutely. Done.